src/ExadelMentorship.Jobs/ReportJob.cs

```diff
@@ -0,0 +1,43 @@
+﻿using ExadelMentorship.BusinessLogic.Interfaces;
+using ExadelMentorship.BusinessLogic.Interfaces.MessageBus;
+using ExadelMentorship.DataAccess;
+using ExadelMentorship.DataAccess.Entities;
+using Hangfire;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+
+namespace ExadelMentorship.Jobs
+{
+    public class ReportJob : BackgroundService
+    {
+        private readonly IMessageProducer _messagePublisher;
+        private IServiceProvider _services;
+
+        public ReportJob(IMessageProducer messagePublisher,IServiceProvider services)
+        {
+            _messagePublisher = messagePublisher;
+            _services = services;
+        }
+        protected async override Task ExecuteAsync(CancellationToken stoppingToken)
+        {
+            using var scope = _services.CreateScope();
+            var all = await scope.ServiceProvider.GetRequiredService<IReportUserRepo>().GetAll();
+            foreach (var item in all)
+            {
+                RecurringJob.AddOrUpdate(
+                    $"ReportSendingJobFor_{item.UserName}",
+                    () => this.Execute(item),
+                    $"0 * {item.Period} * * ?"
+                );
+            }
+        }
+        public async Task Execute(ReportUser reportUser)
```

Comment:
_private_ access-modifier didn't work? If not, u need to explicitly comment why this method is public.